nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/processor/StandardSchedulingContext.java

```diff
@@ -65,7 +66,7 @@ public void leaseControllerService(final String identifier) {
                     + serviceNode.getProxiedControllerService().getIdentifier() + " is in the process of validating its configuration");
         }
 
-        serviceNode.addReference(processorNode);
+        serviceNode.addReference(processorNode, LEASE_TAG);
```

Comment:
can you have a test case for this in order to ensure that we are not leaking this reference?

Reply:
Good point, working on it